### PR TITLE
[OCP] restrict RBAC perms of gpu-operator in OLM bundle

### PR DESCRIPTION
**Pull request by tariq1890**
This PR does the following:

- Remove the wildcard perms found in the ClusterRole defined in the OpenShift OLM bundle
- Remove the unnecessary Role as gpu-operator only needs the ClusterRole
- Parity with the helm chart 

Verified and tested on an OpenShift cluster

### CODE REVIEW

**Review by cdesiniotis — APPROVED** [review]

File: deployments/gpu-operator/templates/clusterrole.yaml [hunk 1]
```diff
@@ -147,10 +147,8 @@ rules:
   - list
   - watch
 - apiGroups:
-  - ""
   - coordination.k8s.io
   resources:
-  - configmaps
```
Comment (shivamerla) [inline deployments/gpu-operator/templates/clusterrole.yaml:153]: why was this removed? did we add this RBAC somewhere else?

Comment (tariq1890) [reply deployments/gpu-operator/templates/clusterrole.yaml:153]: Yeah, this is a duplicate. The configMaps permission already exists in the manifest file

**Review by shivamerla — APPROVED** [review]
